src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -200,76 +188,47 @@ public boolean isCommentNodesRequired() {
 
     @Override
     public void visitToken(DetailAST ast) {
-        final int lineNo = ast.getLineNo();
-        if (illegalLines.get(lineNo) != null
-                && illegalLines.get(lineNo).equals(ast.getColumnNo())) {
-            log(ast, MSG_KEY);
+        final String commentBeginLine = getLine(ast.getLineNo() - 1);
+        final String beforeComment = commentBeginLine.substring(0, ast.getColumnNo());
```

Comment:
`commentBeginLine` and `beforeComment ` are not used elsewhere in this method. I suggest to extract method `isBeforeCommentLegal(ast)` and move logic there

Reply:
Thank you for the suggestion. Method extracted.